dags/sql_examples/sql_check_redshift_etl.py

```diff
@@ -0,0 +1,244 @@
+                row_check = SQLCheckOperator(
+                    task_id=f"yellow_tripdata_row_quality_check_{i}",
+                    sql="sql/row_quality_yellow_tripdata_check.sql",
+                    params=values,
+                )
+            TASK_DICT[f"quality_check_group_{date}"] = quality_check_group
+
+    """
+    #### Run Table-Level Quality Check
+    Ensure that the correct number of rows are present in the table.
+    """
+    value_check = SQLValueCheckOperator(
+        task_id="check_row_count",
+        sql="SELECT COUNT(*) FROM {{ var.json.aws_configs.redshift_table }};",
+        pass_value=20000
+    )
+
+    """
+    #### Run Interval Check
+    Check that the average trip distance today is within a desirable threshold
+    compared to the average trip distance yesterday.
+    """
+    interval_check = SQLIntervalCheckOperator(
+        task_id="check_interval_data",
+        table="{{ var.json.aws_configs.redshift_table }}",
+        days_back=-1,
+        date_filter_column="upload_date",
+        metrics_thresholds={"AVG(trip_distance)": 1.5}
+    )
+
+    """
+    #### Threshold Check
+    Similar to the threshold cases in the Row-Level Check above, ensures that
+    certain row(s) values meet the desired threshold(s).
+    """
+    threshold_check = SQLThresholdCheckOperator(
+        task_id="check_threshold",
+        sql="SELECT MAX(passenger_count) FROM {{ var.json.aws_configs.redshift_table }};",
+        min_threshold=1,
+        max_threshold=8
+    )
+
+    """
+    #### Drop Redshift table
+    Drops the Redshift table if it exists already. This is to make sure that the
+    data in the success and failure cases do not interfere with each other during
+    the data quality check.
+    """
+    drop_redshift_table = PostgresOperator(
+        task_id="drop_table",
+        sql="sql/drop_redshift_yellow_tripdata_table.sql",
+        postgres_conn_id="redshift_default"
+    )
+
+    @task
+    def delete_upload_date(file_path):
+        """
+        #### Drop added column
+        Drops the upload_date column used for this example, as this data is used
+        by other example DAGs in this repository, so it should not interfere
+        with those.
+        """
+        trip_dict = pd.read_csv(
+            file_path,
+            header=0,
+            parse_dates=["pickup_datetime"],
+            infer_datetime_format=True
+        )
+        trip_dict.drop(columns="upload_date", inplace=True)
+        trip_dict.to_csv(
+            file_path,
+            header=True,
+            index=False
+        )
+
+    for date in DATES:
+        TASK_DICT[f"delete_upload_date_{date}"] = delete_upload_date(
+            f"/usr/local/airflow/include/data/yellow_tripdata_sample_{date}.csv"
+        )
+
+    (
+        begin
+        >> [add_upload_date_1, add_upload_date_2]
+        >> converge_1
+        >> [TASK_DICT["upload_2019-01"], TASK_DICT["upload_2019-02"]]
```

Comment:
Same idea as comments in the previous DAG. Seems like it would better to not have to hardcode the `TASK_DICT` keys in the task dependencies but have these dynamically created with the tasks.